src/main/java/uk/ac/ebi/eva/submission/service/SubmissionService.java

```diff
@@ -133,4 +142,27 @@ public void sendMailNotificationForStatusUpdate(SubmissionAccount submissionAcco
         String body = emailHelper.getTextForSubmissionStatusUpdate(submissionAccount, submissionId, submissionStatus, success);
         mailSender.sendEmail(sendTo, subject, body);
     }
+
+    public List<Submission> getSubmissionsByStatus(SubmissionStatus status) {
+
+        return submissionRepository.findByStatus(status.toString());
+
+    }
+
+    public List<SubmissionProcessing> getSubmissionsByProcessingStepAndStatus(SubmissionProcessingStep step,
+                                                                              SubmissionProcessingStatus status) {
+        return submissionProcessingRepository.findByStepAndStatus(step.toString(), status.toString());
+
+    }
+
+    public SubmissionProcessing markSubmissionProcessStepAndStatus(String submissionId,
+                                                                    SubmissionProcessingStep step,
+                                                                    SubmissionProcessingStatus status) {
+        SubmissionProcessing submissionProc = submissionProcessingRepository.findBySubmissionId(submissionId);
```

Comment:
Just to reiterate my understanding, we will have only one entry for submission in the submission_processing_status table and will keep updating the step and its status